### PR TITLE
Added nonce & implemented GTM and CSP changes 

### PR DESCRIPTION
**Pull request by mmmavis**
Made changes based on https://docs.google.com/document/d/1IbfNfn0HUoJ94qDTWiwZjxWmy8ZzisyilGtxOhR5xNc/edit

Related to ticket https://github.com/mozilla/foundation.mozilla.org/issues/9587

### To Review

- It's easier to review this PR without whitespace changes: https://github.com/mozilla/network-pulse/pull/1661/files?w=1
- This is how I test this PR. Please advise if you have a better way to test it. Thanks!
   - run project locally
   - hard refresh the page
   - open dev tool's Network tab, go to `HTML` subtab, find the response with `initiator` type `document`, and look for `nonce-` value in CSP of the response header
   - right click on the page to `view source`, look for `nonce` value
   - verify the two nonce values match
   - if the values match, a few extra `<script>` from GA/GTM should be injected into `<head>`. This can be verified on dev tool's Inspector.
   - repeat step 2 and onward to make sure `nonce` value do change for every new response

### CODE REVIEW

File: js/analytics.js [hunk 1]
```diff
@@ -50,25 +50,7 @@ if (checkDoNotTrack() === false) {
...
-          w[l] = w[l] || [];
-          w[l].push({ "gtm.start": new Date().getTime(), event: "gtm.js" });
-          var f = d.getElementsByTagName(s)[0],
-            j = d.createElement(s),
-            dl = l != "dataLayer" ? "&l=" + l : "";
-          j.async = true;
-          j.src = "https://www.googletagmanager.com/gtm.js?id=" + i + dl;
-          f.parentNode.insertBefore(j, f);
-        })(window, document, "script", "dataLayer", GTM_ID);
-      }
```
Comment (mmmavis) [inline js/analytics.js:71]: Moved to `server.js` so we can wrap it within `<script nonce="..."></script>`

File: js/analytics.js [hunk 2]
```diff
@@ -50,25 +50,7 @@ if (checkDoNotTrack() === false) {
         return identifier;
       };
 
-      const GA_ID = getIdentifier(`ga`);
-      const GTM_ID = getIdentifier(`gtm`);
```
Comment (mmmavis) [inline js/analytics.js:54]: Moved to `server.js`

File: js/security-headers.js [hunk 3]
```diff
@@ -1,47 +1,54 @@
...
-      `https://syndication.twitter.com`,
-      `https://platform.twitter.com`,
-    ],
-    frameAncestors: [`'none'`],
-    manifestSrc: [`'self'`],
-  },
-  reportOnly: false,
-  browserSniff: false,
-};
+export default function (cspNonce) {
```
Comment (mmmavis) [inline js/security-headers.js:4]: Implemented changes based on https://docs.google.com/document/d/1IbfNfn0HUoJ94qDTWiwZjxWmy8ZzisyilGtxOhR5xNc/edit

Note that this is now a function that takes a `cspNonce` parameter that get used for `scriptSrc`

File: server.js [hunk 4]
```diff
@@ -167,14 +179,28 @@ function renderPage(appHtml, reactHelmet, canonicalUrl) {
...
+          var f = d.getElementsByTagName(s)[0],
+            j = d.createElement(s),
+            dl = l != "dataLayer" ? "&l=" + l : "";
+          j.async = true;
+          j.src = "https://www.googletagmanager.com/gtm.js?id=" + i + dl;
+          var n = d.querySelector("[nonce]");
+          n && j.setAttribute("nonce", n.nonce || n.getAttribute("nonce"));
+          f.parentNode.insertBefore(j, f);
+        })(window, document, "script", "dataLayer", "${GTM_ID}");
+        </script>
```
Comment (mmmavis) [inline server.js:203]: This basically is the same script snippet included in the GDoc but in an easier to read format.

**Review by danielfmiranda — APPROVED** [review]
Hi Mavis, 

Thanks for the easy-to-follow testing instructions, as well as the other comments walking through the code changes.

After giving the code a look through and testing on my local machine, I can confirm that there is a new nonce being generated on refresh that it is being passed into the GTM script, and that the google-related scripts are being injected into the head.

Approved! 👍

**Comment by mmmavis** [comment]
@danielfmiranda awesome. Thanks for reading through my long PR description and reviewing this.

**Pull request merged**